pkg/auth/audit/filter.go

```diff
@@ -65,14 +73,29 @@ func (h auditHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 	auditLog, err := newAuditLog(h.auditWriter, req, h.sanitizingRegex)
 	if err != nil {
-		util.ReturnHTTPError(rw, req, 500, err.Error())
+		util.ReturnHTTPError(rw, req, http.StatusInternalServerError, err.Error())
 		return
 	}
 
 	wr := &wrapWriter{ResponseWriter: rw, auditWriter: h.auditWriter, statusCode: http.StatusOK}
 	h.next.ServeHTTP(wr, req)
 
-	auditLog.write(user, req.Header, wr.Header(), wr.statusCode, wr.buf.Bytes())
+	err = auditLog.write(user, req.Header, wr.Header(), wr.statusCode, wr.buf.Bytes())
+	if err == nil {
+		return
+	}
+
+	// Locking after next is called to avoid performance hits on the request.
```

Comment:
This comment doesn't make sense to me - next isn't called below that I can see.

Reply:
Never mind - misread the comment. Is a bit weird that next is called and then we lock w/ comment ~8 lines later, but that's not a big deal, and the misread was more on my failure to read carefully.